### PR TITLE
Update requirements for OptunaHub in README

### PR DESCRIPTION
**Pull request by y0z**
## Motivation & Description of the changes

Add required dependencies for AutoSampler.

### CODE REVIEW

**Comment by y0z** [comment]
@porink0424 @kAIto47802 Could you review this PR?

File: README.md [hunk 1]
```diff
@@ -206,6 +206,7 @@ You can use the registered features and publish your packages.
 
 ```shell
 $ pip install optunahub
+$ pip install cmaes scipy torch # install AutoSampler dependencies
```
Comment (nabenabe0928) [inline README.md:209]:
```suggestion
# Install AutoSampler dependencies (CPU only is sufficient for PyTorch)
$ pip install cmaes scipy torch --extra-index-url https://download.pytorch.org/whl/cpu
```

**Review by nabenabe0928 — COMMENTED** [review]
I think it would be more friendly if the example installs the CPU-only version.

File: README.md (hunk 1 above)
Comment (y0z) [reply README.md]: Applied.

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Comment by kAIto47802** [comment]
As stated in [the pip documentation](https://pip.pypa.io/en/stable/cli/pip_install/#finding-packages), there is no prioritization between PyPI and locations added with `--extra-index-url`. This might makes it non-obvious that the CPU version is prioritized over the default GPU version when installed on Linux. However, experimental confirmation shows that the CPU version is indeed installed in such cases.

**Review by kAIto47802 — APPROVED** [review]
LGTM

**Review by porink0424 — APPROVED** [review]
LGTM!

**Pull request merged**